Disable "Back" button on the first step of the setup

ArjenMiedema opened the pull request:
### Description
When you enter the setup of Magento, the "Back" button is enabled
on the first step. When you click this, you go straight to the
final step of the installation, without entering any data in the
steps, so you can't install Magento at all.

With this commit the back button on the first step of the setup
is always disabled, since there's no previous step, except for the
page where you agree with the terms and conditions, which is not
part of the wizard.

### Fixed Issues
1. magento/magento2#14307: Possible to press the "Previous" button while in the first step of the installation

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. Create a fresh Magento 2 environment
2. Go to the setup wizard in your browser
3. When on the first step, try to use the "Back" button. This is now disabled with this fix.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
Disable "Back" button on the first step of the setup

When you enter the setup of Magento, the "Back" button is enabled
on the first step. When you click this, you go straight to the
final step of the installation, without entering any data in the
steps, so you can't install Magento at all.

With this commit the back button on the first step of the setup
is always disabled, since there's no previous step, except for the
page where you agree with the terms and conditions, which is not
part of the wizard.

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=14309) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=14309) before we can accept your contribution.<br/>

Commit:
Remove "goToStart" method because it's no longer used

The "goToStart" method was used for the "Back" button on the first
step, but since that button is disabled now, the method is no longer
used within Magento. Therefore it is removed from the codebase now.

mzeis reviewed (approved):
Hi @ArjenMiedema, thanks for your PR! It works for me. 👍

mzeis reviewed (approved):
Approving again as automatic process wasn't triggered.

magento-engcom-team commented:
Hi @mzeis, thank you for the review.
[ENGCOM-1148](https://jira.corp.magento.com/browse/ENGCOM-1148) has been created to process this Pull Request

magento-engcom-team commented:
@ArjenMiedema thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

mzeis commented:
Hi @ArjenMiedema, thank you for your PR! It has been merged into `2.2-develop`.
